File: backend/HRManagementSystem/src/Core/HRManagementSystem.Domain/Entities/EmployeeAllowance.cs
using HRManagementSystem.Domain.Enums;

namespace HRManagementSystem.Domain.Entities;

public class EmployeeAllowance
{
    public int EmployeeId { get; set; }
    public int AllowanceId { get; set; }
    public decimal? Amount { get; set; }       
    public bool? IsPercentage { get; set; }
    public RecurrenceType Recurrence { get; set; } = RecurrenceType.OneTime;
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public Employee Employee { get; set; }
    public Allowance Allowance { get; set; }
}

File: frontend/src/app/models/employee-allowance.ts
export enum RecurrenceType {
    OneTime = 1,
    Period = 2,
    Annual = 3,
    Permanent = 4,
}

export interface EmployeeAllowanceDto {
    employeeId: number;
    allowanceId: number;
    employeeName?: string;
    allowanceName?: string;
    amount?: number;
    isPercentage?: boolean;
    effectiveDate?: string; // Kept for backward compatibility if needed, but new fields are primary
    recurrence: RecurrenceType;
    startDate?: string;
    endDate?: string;
}

export interface CreateEmployeeAllowanceDto {
    employeeId: number;
    allowanceId: number;
    amount?: number;
    isPercentage?: boolean;
    recurrence: RecurrenceType;
    startDate?: string;
    endDate?: string;
}

export interface UpdateEmployeeAllowanceDto {
    amount?: number;
    isPercentage?: boolean;
    recurrence: RecurrenceType;
    startDate?: string;
    endDate?: string;
}
